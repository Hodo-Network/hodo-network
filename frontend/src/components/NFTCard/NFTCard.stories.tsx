import React from "react";
import { Story, Meta } from "@storybook/react";

import { NFTCard, NFTCardProps } from "./NFTCard";

export default {
  title: "Components/NFTCard",
  component: NFTCard,
} as Meta;

const Template: Story<NFTCardProps> = (args) => <NFTCard {...args} />;

export const Default = Template.bind({});
Default.args = {};
