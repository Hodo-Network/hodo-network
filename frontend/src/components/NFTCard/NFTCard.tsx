import React from "react";
import { NavLink } from "react-router-dom";
// import { Web3Provider } from "@ethersproject/providers";
// import { useWeb3React } from "@web3-react/core";
import { NFT } from "../../typings/nft";
// import { NATIVE_CURRENCY } from "../../constants";
import { ROUTE_COLLECTIONS } from "../../constants/routes";
import RarityBadge from "../RarityBadge";

export interface NFTCardProps {
  nft: NFT;
  className?: string;
}

export const NFTCard: React.FC<NFTCardProps> = ({
  nft: { id, name, category, image, data, price },
  className,
}) => {
  // const { chainId } = useWeb3React<Web3Provider>();

  return (
    <NavLink
      to={`${ROUTE_COLLECTIONS}/${category}/${id}`}
      className={`${className} flex flex-col rounded-md shadow overflow-hidden transition-transform duration-300 transform hover:-translate-y-1`}>
      <div className='flex-shrink-0'>
        <img className='h-54 w-full object-scale-down' src={image} alt={name} />
      </div>
      <div className='flex-1 bg-white dark:bg-gray-800 p-4 flex flex-col justify-between'>
        <div className='flex-1'>
          <p className='text-base font-semibold text-gray-900 dark:text-gray-100'>
            {name}
          </p>
          <p className='mt-1 text-sm text-gray-600 dark:text-gray-300'>
            {data.description}
          </p>
        </div>
        <div className='mt-6 flex items-center justify-between'>
          <div className='flex-shrink-0'>
            <RarityBadge rarity={data.rarity} />
          </div>
          <div className='ml-3 text-sm font-medium text-green-600 dark:text-green-500'>
            {price.value} {price.units}
            {/* {(chainId && NATIVE_CURRENCY[chainId]) || NATIVE_CURRENCY[0]} */}
          </div>
        </div>
      </div>
    </NavLink>
  );
};
