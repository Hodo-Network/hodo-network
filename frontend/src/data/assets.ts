import { NFT } from '../typings/nft';
// https://www.latlong.net/
// https://gateway.pinata.cloud/ipfs/QmWdDjeh7R5K9w7uzh7bSMgmuDoNTMYXVY5ER69LdHwWc4

export const assets: Array<NFT> = [
    {
        id: '1000',
        tokenId: '0',
        tokenURI: '',
        name: "Cape Town",
        description: 'Capital City of South Africa.',
        image: "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: -33.924870,
        lng: 18.424055,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        creator: '0x000',
        supply: '1',
        amount: '100',
        price: 2,
        created_at: '1634233032',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1001',
        tokenId: '1',
        tokenURI: '',
        name: "Bangkok",
        description: 'Capital City of Thailand.',
        image: "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: 13.756331,
        lng: 100.501762,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1002',
        tokenId: '2',
        tokenURI: '',
        name: "Jakarta",
        description: 'Capital City of Indonesia.',
        image: "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: -6.175110,
        lng: 106.865036,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1003',
        tokenId: '3',
        tokenURI: '',
        name: "Guatemala City",
        description: 'Capital City of Guatemala.',
        image: "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: 14.634915,
        lng: -90.506882,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1004',
        tokenId: '4',
        tokenURI: '',
        name: "Jamestown",
        description: 'Capital City of Saint Helena.',
        image: "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: 36.434269,
        lng: -84.935371,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1005',
        tokenId: '5',
        tokenURI: '',
        name: "Afghanistan",
        description: 'Here is an NFT description 1.',
        image: "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: 33.93911,
        lng: 67.709953,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1006',
        tokenId: '6',
        tokenURI: '',
        name: "Afghanistan",
        description: 'Here is an NFT description 2.',
        image: "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: 33.93911,
        lng: 67.709953,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1007',
        tokenId: '7',
        tokenURI: '',
        name: "Afghanistan",
        description: 'Here is an NFT description 3.',
        image: "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: 33.93911,
        lng: 67.709953,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1008',
        tokenId: '8',
        tokenURI: '',
        name: "Andorra",
        description: 'Here is an NFT description 4.',
        image: "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: 42.506285,
        lng: 1.521801,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1009',
        tokenId: '9',
        tokenURI: '',
        name: "Australia",
        description: 'Here is an NFT description 5.',
        image: "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: -25.274398,
        lng: 133.775136,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1010',
        tokenId: '10',
        tokenURI: '',
        name: "Argentina",
        description: 'Here is an NFT description 6.',
        image: "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        lat: -38.416097,
        lng: -63.616672,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
];