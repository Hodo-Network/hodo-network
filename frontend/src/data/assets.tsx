import { NFT } from '../typings/nft';
// https://www.latlong.net/
// https://gateway.pinata.cloud/ipfs/QmWdDjeh7R5K9w7uzh7bSMgmuDoNTMYXVY5ER69LdHwWc4

export const assets: Array<NFT> = [
    {
        id: '1000',
        tokenId: '0',
        tokenURI: '',
        name: "Cape Town",
        description: 'Capital City of South Africa.',
        image: "/assets/images/skyline.jpeg",
        lat: -33.924870,
        lng: 18.424055,
        contractAddress: '0x0000000000000000000000000000000000000000',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        creator: '0x000',
        supply: '1',
        amount: '100',
        price: 2,
        created_at: '1634233032',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1001',
        tokenId: '1',
        tokenURI: '',
        name: "Bangkok",
        description: 'Capital City of Thailand.',
        image: "/assets/images/skyline.jpeg",
        lat: 13.756331,
        lng: 100.501762,
        contractAddress: '0x0000000000000000000000000000000000000002',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1002',
        tokenId: '2',
        tokenURI: '',
        name: "Jakarta",
        description: 'Capital City of Indonesia.',
        image: "/assets/images/skyline.jpeg",
        lat: -6.175110,
        lng: 106.865036,
        contractAddress: '0x0000000000000000000000000000000000000002',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1003',
        tokenId: '3',
        tokenURI: '',
        name: "Guatemala City",
        description: 'Capital City of Guatemala.',
        image: "/assets/images/skyline.jpeg",
        lat: 14.634915,
        lng: -90.506882,
        contractAddress: '0x0000000000000000000000000000000000000002',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1004',
        tokenId: '4',
        tokenURI: '',
        name: "Jamestown",
        description: 'Capital City of Saint Helena.',
        image: "/assets/images/skyline.jpeg",
        lat: 36.434269,
        lng: -84.935371,
        contractAddress: '0x0000000000000000000000000000000000000002',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1005',
        tokenId: '5',
        tokenURI: '',
        name: "Afghanistan",
        description: 'Here is an NFT description 1.',
        image: "/assets/images/forest.jpeg",
        lat: 33.93911,
        lng: 67.709953,
        contractAddress: '0x0000000000000000000000000000000000000001',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1006',
        tokenId: '6',
        tokenURI: '',
        name: "Afghanistan",
        description: 'Here is an NFT description 2.',
        image: "/assets/images/forest.jpeg",
        lat: 33.93911,
        lng: 67.709953,
        contractAddress: '0x0000000000000000000000000000000000000001',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1007',
        tokenId: '7',
        tokenURI: '',
        name: "Afghanistan",
        description: 'Here is an NFT description 3.',
        image: "/assets/images/forest.jpeg",
        lat: 33.93911,
        lng: 67.709953,
        contractAddress: '0x0000000000000000000000000000000000000001',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1008',
        tokenId: '8',
        tokenURI: '',
        name: "Andorra",
        description: 'Here is an NFT description 4.',
        image: "/assets/images/forest.jpeg",
        lat: 42.506285,
        lng: 1.521801,
        contractAddress: '0x0000000000000000000000000000000000000001',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1009',
        tokenId: '9',
        tokenURI: '',
        name: "Australia",
        description: 'Here is an NFT description 5.',
        image: "/assets/images/forest.jpeg",
        lat: -25.274398,
        lng: 133.775136,
        contractAddress: '0x0000000000000000000000000000000000000001',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
    {
        id: '1010',
        tokenId: '10',
        tokenURI: '',
        name: "Argentina",
        description: 'Here is an NFT description 6.',
        image: "/assets/images/forest.jpeg",
        lat: -38.416097,
        lng: -63.616672,
        contractAddress: '0x0000000000000000000000000000000000000001',
        contractName: 'Contract name',
        contractVerified: 0,
        owner_address: '0x',
        created_at: '1634233032',
        supply: '1',
        amount: '100',
        price: 2,
        creator: '0x000',
        tx_hash: '0x0000',
        is_deleted: 0,
        sold: 0,
    },
];