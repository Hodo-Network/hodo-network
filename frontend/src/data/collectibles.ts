import { NFT, NFTCategory, NFTRarity, Vendor } from '../typings/nft';
// https://www.latlong.net/

export const collectibles: Array<NFT> = [
    {
        id: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ1',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ1',
        // activeOrderId: null,
        owner: '0x',
        name: "Cape Town",
        category: NFTCategory.CAPITAL,
        image: "https://gateway.pinata.cloud/ipfs/QmWdDjeh7R5K9w7uzh7bSMgmuDoNTMYXVY5ER69LdHwWc4",
        data: {
            description: 'Capital City of South Africa.',
            rarity: NFTRarity.EPIC,
            location: {
                lat: -33.924870,
                long: 18.424055,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 1,
            units: "AVAX"
        },
    },
    {
        id: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ2',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ2',
        // activeOrderId: null,
        owner: '0x',
        name: "Bangkok",
        category: NFTCategory.CAPITAL,
        image:
            "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Capital City of Thailand.',
            rarity: NFTRarity.RARE,
            location: {
                lat: 13.756331,
                long: 100.501762,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 1,
            units: "AVAX"
        },
    },
    {
        id: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ3',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ3',
        // activeOrderId: null,
        owner: '0x',
        name: "Jakarta",
        category: NFTCategory.CAPITAL,
        image:
            "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Capital City of Indonesia.',
            rarity: NFTRarity.RARE,
            location: {
                lat: -6.175110,
                long: 106.865036,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 1,
            units: "AVAX"
        },
    },
    {
        id: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ4',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ4',
        // activeOrderId: null,
        owner: '0x',
        name: "Guatemala City",
        category: NFTCategory.CAPITAL,
        image:
            "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Capital City of Guatemala.',
            rarity: NFTRarity.COMMON,
            location: {
                lat: 14.634915,
                long: -90.506882,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 0.5,
            units: "AVAX"
        },
    },
    {
        id: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ5',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ5',
        // activeOrderId: null,
        owner: '0x',
        name: "Jamestown",
        category: NFTCategory.CAPITAL,
        image:
            "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Capital City of Saint Helena.',
            rarity: NFTRarity.COMMON,
            location: {
                lat: 36.434269,
                long: -84.935371,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 0.5,
            units: "AVAX"
        },
    },
    {
        id: 'QmQBSaJvXv3SfYMpEEyVzhtUCquhTFxvo5F7gvhLRGvMuB',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'QmQBSaJvXv3SfYMpEEyVzhtUCquhTFxvo5F7gvhLRGvMuB',
        // activeOrderId: null,
        owner: '0x',
        name: "Afghanistan",
        category: NFTCategory.COUNTRY,
        image:
            "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Here is an NFT description 1.',
            rarity: NFTRarity.COMMON,
            location: {
                lat: 33.93911,
                long: 67.709953,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 0.25,
            units: "AVAX"
        },
    },
    {
        id: 'QmWbUonVdzNyCMqAdFC2URpc9GXV5NwdeHbitxDcRzydfY',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'QmWbUonVdzNyCMqAdFC2URpc9GXV5NwdeHbitxDcRzydfY',
        // activeOrderId: null,
        owner: '0x',
        name: "Afghanistan",
        category: NFTCategory.COUNTRY,
        image:
            "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Here is an NFT description 2.',
            rarity: NFTRarity.EPIC,
            location: {
                lat: 33.93911,
                long: 67.709953,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 1,
            units: "AVAX"
        },
    },
    {
        id: 'QmaLAahk2f1Qyvtz171thg85ArETnXyZm47fyepNvy9bnP',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'QmaLAahk2f1Qyvtz171thg85ArETnXyZm47fyepNvy9bnP',
        // activeOrderId: null,
        owner: '0x',
        name: "Afghanistan",
        category: NFTCategory.COUNTRY,
        image:
            "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Here is an NFT description 3.',
            rarity: NFTRarity.RARE,
            location: {
                lat: 33.93911,
                long: 67.709953,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 1,
            units: "AVAX"
        },
    },
    {
        id: 'QmZ8ArvpTy9J4D9gMzsPgitwBm2NrGFhkQDBkEAhRFDAxp',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'QmZ8ArvpTy9J4D9gMzsPgitwBm2NrGFhkQDBkEAhRFDAxp',
        // activeOrderId: null,
        owner: '0x',
        name: "Andorra",
        category: NFTCategory.COUNTRY,
        image:
            "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Here is an NFT description 4.',
            rarity: NFTRarity.RARE,
            location: {
                lat: 42.506285,
                long: 1.521801,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 1,
            units: "AVAX"
        },
    },
    {
        id: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ11',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ11',
        // activeOrderId: null,
        owner: '0x',
        name: "Australia",
        category: NFTCategory.COUNTRY,
        image:
            "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Here is an NFT description 5.',
            rarity: NFTRarity.COMMON,
            location: {
                lat: -25.274398,
                long: 133.775136,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 0.5,
            units: "AVAX"
        },
    },
    {
        id: 'QmQWmYPT9xGgTwDS97q6JwPoBagKcGNYoaih2x8cArRPLm',
        contractAddress: '0x0000000000000000000000000000000000000000',
        // tokenId: 'QmQWmYPT9xGgTwDS97q6JwPoBagKcGNYoaih2x8cArRPLm',
        // activeOrderId: null,
        owner: '0x',
        name: "Argentina",
        category: NFTCategory.COUNTRY,
        image:
            "https://images.unsplash.com/photo-1574158676804-4d719c7eb7c0?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        data: {
            description: 'Here is an NFT description 6.',
            rarity: NFTRarity.COMMON,
            location: {
                lat: -38.416097,
                long: -63.616672,
            },
        },
        vendor: Vendor.HODO_NETWORK,
        price: {
            value: 0.25,
            units: "AVAX"
        },
    },
];