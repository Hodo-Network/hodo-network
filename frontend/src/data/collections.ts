import { Collection } from '../typings/nft';

export const collections: Array<Collection> = [
    {
        id: 'BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ1',
        name: 'Continents',
        description: '',
        thumbnail: "https://images.unsplash.com/flagged/photo-1585324853527-1c567d53bb72?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=800",
        contractAddress: "0x00",
        network: 43113,
        verified: 0
    },
    {
        id: 'RFkpHV9KcMtjxL19wsJ1BZTDWRaK6rQmnSLrrq4PANBeueYfD',
        name: 'Countries',
        description: '',
        thumbnail: "ipfs://QmQBSaJvXv3SfYMpEEyVzhtUCquhTFxvo5F7gvhLRGvMuB",
        contractAddress: "0x00",
        network: 43113,
        verified: 0
    },
    {
        id: 'RFkpHV9KcMtjxL19wsJ1BZTDWRaK6rQmnSLrrq4PANBeueYfD',
        name: 'Capitals',
        description: '',
        thumbnail: "https://images.unsplash.com/photo-1452838269443-c86e6a3134f9?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        contractAddress: "0x00",
        network: 43113,
        verified: 0
    },
    {
        id: 'RFkpHV9KcMtjxL19wsJ1BZTDWRaK6rQmnSLrrq4PANBeueYfD',
        name: 'Monuments',
        description: '',
        thumbnail: "https://images.unsplash.com/photo-1565178646088-9b390ea73981?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=512&q=80",
        contractAddress: "0x00",
        network: 43113,
        verified: 0
    }
];
