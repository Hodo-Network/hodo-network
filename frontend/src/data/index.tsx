import { collectibles } from "./collectibles";
import { collections } from "./collections";
import { user } from "./user";

export { collectibles, collections, user };
