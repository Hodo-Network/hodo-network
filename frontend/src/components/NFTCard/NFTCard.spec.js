import { cleanup } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import { create } from "react-test-renderer";
import NFTCard from "./index";
import { collectibles } from "../../data/collectibles";

const nft = collectibles[0];

afterEach(cleanup);

test("renders correctly", () => {
  let tree = create(
    <Router initialEntries={["/"]}>
      <NFTCard nft={nft} />
    </Router>
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
