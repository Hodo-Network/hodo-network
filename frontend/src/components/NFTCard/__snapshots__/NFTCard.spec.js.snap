// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  aria-current={null}
  className="undefined flex flex-col rounded-md shadow overflow-hidden transition-transform duration-300 transform hover:-translate-y-2"
  href="/collections/capital/BZTDWRaK6rQmnSLrrq4PANBeueYfDRFkpHV9KcMtjxL19wsJ1"
  onClick={[Function]}
>
  <div
    className="flex-shrink-0"
  >
    <img
      alt="Cape Town"
      className="h-54 w-full object-scale-down"
      src="https://gateway.pinata.cloud/ipfs/QmWdDjeh7R5K9w7uzh7bSMgmuDoNTMYXVY5ER69LdHwWc4"
    />
  </div>
  <div
    className="flex-1 bg-white dark:bg-gray-800 p-4 flex flex-col justify-between"
  >
    <div
      className="flex-1"
    >
      <p
        className="text-base font-semibold text-gray-900 dark:text-gray-100"
      >
        Cape Town
      </p>
      <p
        className="mt-1 text-sm text-gray-600 dark:text-gray-300"
      >
        Capital City of South Africa.
      </p>
    </div>
    <div
      className="mt-6 flex items-center justify-between"
    >
      <div
        className="flex-shrink-0"
      >
        <span
          className="inline-flex items-center px-2.5 py-1 rounded-md text-sm font-medium capitalize bg-yellow-100 text-yellow-800 undefined"
        >
          epic
        </span>
      </div>
      <div
        className="ml-3 text-sm font-medium text-green-600 dark:text-green-500"
      >
        1
         
        AVAX
      </div>
    </div>
  </div>
</a>
`;
